plutus-playground-client/src/Action/Validation.purs

```diff
@@ -0,0 +1,34 @@
+module Action.Validation
+  ( actionIsValid
+  , actionsAreValid
+  ) where
+
+import Data.Array as Array
+import Data.Lens (view)
+import Playground.Types (ContractCall(..), SimulatorWallet)
+import Prelude ((==), (&&), (<<<))
+import Schema.Types (FormArgument)
+import Types (_simulatorWalletWallet, _walletId)
+import Validation (isValid)
+import Wallet.Emulator.Wallet (Wallet)
+
+actionIsValid :: Array SimulatorWallet -> ContractCall FormArgument -> Boolean
+actionIsValid simulatorWallets simulatorAction = actionWalletsExist simulatorAction && isValid simulatorAction
+  where
+  walletExists :: Wallet -> Boolean
+  walletExists wallet =
+    Array.any
+      (\simulatorWallet -> view _walletId wallet == view (_simulatorWalletWallet <<< _walletId) simulatorWallet)
+      simulatorWallets
+
+  actionWalletsExist :: ContractCall FormArgument -> Boolean
+  actionWalletsExist (AddBlocks _) = true -- because there is no wallet to check for in the first place
+
+  actionWalletsExist (AddBlocksUntil _) = true -- ditto
+
+  actionWalletsExist (CallEndpoint a@{ caller }) = walletExists caller
+
+  actionWalletsExist (PayToWallet a@{ sender, recipient }) = walletExists sender && walletExists recipient
+
+actionsAreValid :: Array SimulatorWallet -> Array (ContractCall FormArgument) -> Boolean
+actionsAreValid simulatorWallets simulatorActions = Array.all (actionIsValid simulatorWallets) simulatorActions
```

Comment:
You could eta reduce this to:

```purescript
actionsAreValid simulatorWallets = Array.all (actionIsValid simulatorWallets)
```

...or even:

```purescript
actionsAreValid = Array.all <<< actionIsValid
```

But when you look at it like that, you have to wonder if it's worth its own function?

Reply:
Good point. I'll get rid of it.